### ISSUE TITLE
Restore deleted file

### ISSUE THREAD

**Issue by lunanigra** [rtyley/bfg-repo-cleaner#245]
Hello, I found following page...
https://stackoverflow.com/questions/31142105/possible-to-undo-permanent-deletion-of-file

Unluckily I do not have a mirror; but a normal clone containing the old file. Means, "git cat-file -p xxx" shows the original content. But "git update-ref" does not work.

fatal: update_ref failed for ref 'refs/heads/develop': cannot update the ref 'refs/heads/develop': Trying to write non-commit object xxx to branch refs/heads/develop

**Comment by javabrett** [comment]
This appears unrelated to BFG.  You should probably ask this question on StackOverflow.

But if you have access to the raw file content, what's to stop you from just copying the file into your repo and committing it?

**Comment by lunanigra** [comment]
This idea is restoring the file incl. history - not only re-commit the latest version. Perhaps not possible at all...

**Comment by javabrett** [comment]
Yeah that's going to be hard.  I expect you'd need to do something like run BFG or `filter-branch` again and regenerate the whole repo if you want the history to appear as and when it occurred.  You might be able to replay a history of just that one file as commits on-top of your current `HEAD`, if you don't mind new commits.
